CMakeLists.txt

```diff
@@ -0,0 +1,69 @@
+cmake_minimum_required(VERSION 3.2)
+project(earcut_hpp)
+set (INCLUDE_DIRS "")
+
+#setup compiler flags
+if(MSVC)
+    if(CMAKE_CXX_FLAGS MATCHES "/W[0-4]")
+        string(REGEX REPLACE "/W[0-4]" "/W3" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+    else()
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /W3")
+    endif()
+elseif(CMAKE_COMPILER_IS_GNUCC OR CMAKE_COMPILER_IS_GNUCXX)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pipe -std=c++14 -Werror")
+    # some flags that may be used in the future: -pedantic -Wconversion -Wall
+    if(CMAKE_BUILD_TYPE EQUAL "DEBUG")
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O0")
+        # optional: -fsanitize=signed-integer-overflow,undefined
+    else()
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O3")
+        # optional: -march=native (builds with the optimizations available on the build machine (only for local use!))
+    endif()
+else()
+    message(WARNING "Building earcut with unknown compiler.")
```

Comment:
This message is triggered when compiling with Clang. It means that Clang won't receive any compiler flags.